From 5f3d7babde956118806f95441ec917629718e59f Mon Sep 17 00:00:00 2001
From: Sonali Naresh <sonanare@qti.qualcomm.com>
Date: Thu, 9 Jul 2026 12:10:16 +0530
Subject: [PATCH] WFA fix for wps pbc issue

Observing wps failure since EndPoint was not being set
Fix is to add the End Points during dev_reset_default

Signed-off-by: Sonali Naresh <sonanare@qti.qualcomm.com>
---
 agent/src/beerocks/slave/backhaul_manager/backhaul_manager.cpp | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/agent/src/beerocks/slave/backhaul_manager/backhaul_manager.cpp b/agent/src/beerocks/slave/backhaul_manager/backhaul_manager.cpp
index cd6cf52..dbbcb7e 100644
--- a/agent/src/beerocks/slave/backhaul_manager/backhaul_manager.cpp
+++ b/agent/src/beerocks/slave/backhaul_manager/backhaul_manager.cpp
@@ -4198,6 +4198,9 @@ void BackhaulManager::handle_dev_reset_default(
             LOG(INFO) << "Setting Invalid MLDUnit for EndPoint on Radio "
                       << radio_info->hostap_iface;
             radio_info->sta_wlan_hal->update_mld_unit(DISABLED_MLDUNIT);
+            // Enable WiFi.EndPoint.*.Enable=1 for all radios (PrplOS/WHM path only)
+            LOG(INFO) << "Enabling WiFi EndPoint (Enable=1) for radio " << radio_info->hostap_iface;
+            radio_info->sta_wlan_hal->enable_disable_ep(true);
         }
     }
 
-- 
2.34.1

